src/FormMigration/FormMetaDecorator.php

```diff
@@ -446,4 +446,67 @@ public function getFormFieldsPlacement(): string
     {
         return give_get_meta($this->form->id, '_give_ffm_placement', true);
     }
+
+    /**
+     * @unreleased
+     */
+    public function getFeeRecoverySettings(): array
+    {
+        $feeRecoveryStatus = $this->getMeta('_form_give_fee_recovery');
+
+        if ($feeRecoveryStatus === 'disabled') {
+            return [];
+        }
```

Comment:
I would reverse this and check specifically if it's enabled.

```
give_is_setting_enabled( give_get_meta( $form_id, '_form_give_fee_recovery', true );
```

If it's set to global we should just let the global/default GiveWP fee recovery setting determine if it should add the block or not.

Reply:
I intended to implement an early return if Fee Recovery is disabled then to preserve all existing options in the new form, even if it is configured to use global settings.

Reply:
I would just return early if it's not enabled.  I don't see the point in reaching for meta if it's set to global.

Reply:
Could you confirm that https://github.com/impress-org/givewp/pull/7025/commits/c627f0120cb8ecc94237859c7fc26a25218fd425 is correct?